portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -6567,10 +6568,14 @@ protected void reindex(long[] userIds) throws SearchException {
 	}
 
 	protected void reindex(User user) throws SearchException {
-		Indexer<User> indexer = IndexerRegistryUtil.nullSafeGetIndexer(
-			User.class);
+		Indexer<User> indexer = IndexerRegistryUtil.getIndexer(User.class);
 
-		indexer.reindex(user);
+		if (indexer == null) {
+			_unindexedUserQueue.add(user);
```

Comment:
It must be reviewed yet because it will keep adding in some cases where the portal is already started. Search team can continue improving this solution.

Reply:
Or activated will keep been call and it wont cause issues